api4/user_local.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api4
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+func (api *API) InitUserLocal() {
+	api.BaseRoutes.Users.Handle("", api.ApiLocal(getUsers)).Methods("GET")
+	api.BaseRoutes.Users.Handle("/ids", api.ApiLocal(getUsersByIds)).Methods("POST")
+	api.BaseRoutes.User.Handle("", api.ApiLocal(getUser)).Methods("GET")
+
+	api.BaseRoutes.UserByUsername.Handle("", api.ApiLocal(localGetUserByUsername)).Methods("GET")
+	api.BaseRoutes.UserByEmail.Handle("", api.ApiLocal(localGetUserByEmail)).Methods("GET")
+}
+
+func localGetUserByUsername(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUsername()
+	if c.Err != nil {
+		return
+	}
+
+	user, err := c.App.GetUserByUsername(c.Params.Username)
+	if err != nil {
+		return
+	}
+
+	etag := user.Etag(*c.App.Config().PrivacySettings.ShowFullName, *c.App.Config().PrivacySettings.ShowEmailAddress)
+
+	if c.HandleEtag(etag, "Get User", w, r) {
+		return
+	}
+
+	if c.App.Session().UserId == user.Id {
```

Comment:
Same as above - AFAIK, `c.App.Session().UserId` is always empty string in local mode.

Reply:
Yep, this should only take into account the `else` case. `IsSystemAdmin` will always shortcircuit to `true`, but I'd rather keep it in case we do any refactor in the future.